src/NBitcoin/BlockStake.cs

```diff
@@ -270,16 +270,57 @@ public override Transaction CreateTransaction(byte[] bytes)
         }
     }
 
+    public interface ISmartContractBlockHeader
+    {
+        uint256 HashStateRoot { get; set; }
+
+        uint256 ReceiptRoot { get; set; }
+
+        Bloom LogsBloom { get; set; }
+    }
+
     /// <summary>
     /// A POS block header, this will create a work hash based on the X13 hash algos.
     /// </summary>
 #pragma warning disable 618
-    public class PosBlockHeader : BlockHeader
+    public class PosBlockHeader : BlockHeader, ISmartContractBlockHeader
 #pragma warning restore 618
     {
+        // Determines whether this object should serialize the new fields associated with smart contracts.
+        public bool HasSmartContractFields => (this.version & 0x10000000 /* ExtendedHeaderBit */) != 0;
```

Comment:
Can we not reference the extended header bit from the ThresholdConditionCache? Or vice versa?

Reply:
Not from here but I've moved the definition.

Reply:
Not from here but I've moved the definition.